Pear/SceneDelegate.swift

```diff
@@ -29,32 +29,18 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 
         if let signIn = GIDSignIn.sharedInstance(), signIn.hasPreviousSignIn() {
             signIn.restorePreviousSignIn()
-
-            let didCompleteOnboarding = UserDefaults.standard.bool(forKey: Constants.UserDefaults.onboardingCompletion)
-            let refreshToken = UserDefaults.standard.string(forKey: Constants.UserDefaults.refreshToken)
-            let rootVC = didCompleteOnboarding ?
-                HomeViewController() :
-                OnboardingPageViewController(transitionStyle: .scroll, navigationOrientation: .horizontal)
-
-            guard let unwrappedToken = refreshToken else {
-                navigationController.pushViewController(LoginViewController(), animated: false)
-                window.rootViewController = navigationController
-                return
-            }
-            
-            NetworkManager.shared.refreshUserToken(token: unwrappedToken).observe { result in
-                DispatchQueue.main.async {
-                    switch result {
-                    case .value(let response):
-                        let userSession = response.data
-                        UserDefaults.standard.set(userSession.accessToken, forKey: Constants.UserDefaults.accessToken)
-                        UserDefaults.standard.set(userSession.refreshToken, forKey: Constants.UserDefaults.refreshToken)
-                        navigationController.pushViewController(rootVC, animated: false)
-                        window.rootViewController = navigationController
-                    case .error:
-                        navigationController.pushViewController(LoginViewController(), animated: false)
-                        window.rootViewController = navigationController
-                    }
+            Networking2.validateAccessToken() { success in
+                if success {
+                    let didCompleteOnboarding = UserDefaults.standard.bool(forKey: Constants.UserDefaults.onboardingCompletion)
```

Comment:
we took care of this on the backend for you, is there a reason to send back whether or not a user has onboarded then or is it fine to just maintain that information internally? I think caching it here is probably the better idea since you need to validate the access token and then get the current user (2 network calls) versus what you're doing here and caching.

also, i'm assuming that you update the user's `has_onboarded` somewhere else since we don't take care of that on our end